balancer/roundrobin/roundrobin.go

```diff
@@ -69,15 +69,13 @@ type rrPicker struct {
 	// created. The slice is immutable. Each Get() will do a round robin
 	// selection from it and return the selected SubConn.
 	subConns []balancer.SubConn
-
-	mu   sync.Mutex
-	next int
+	next     uint32
 }
 
 func (p *rrPicker) Pick(balancer.PickInfo) (balancer.PickResult, error) {
-	p.mu.Lock()
-	sc := p.subConns[p.next]
-	p.next = (p.next + 1) % len(p.subConns)
-	p.mu.Unlock()
+	subConnsLen := uint32(len(p.subConns))
+	consumed := atomic.AddUint32(&p.next, 1)
```

Comment:
Nit: I don't like the name consumed here. s/next || nextIndex or something along those lines.

Reply:
yes, i agree with you

Reply:
Thanks for switching.